api.go

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2025 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2025 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package modusdb
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/dgraph/v24/dql"
+	"github.com/dgraph-io/dgraph/v24/schema"
+	"github.com/hypermodeinc/modusdb/api/apiutils"
+	"github.com/hypermodeinc/modusdb/api/structreflect"
+)
+
+func Create[T any](db *DB, object T, ns ...uint64) (uint64, T, error) {
```

Comment:
This pattern will become a problem as we may need other options in the future.  I suggest this use the functional options pattern instead, ie, `WithNamespace(uint64)` - which is more idiomatic for Go anyway.